src/components/button-swap/button-swap.stories.mdx

```diff
@@ -0,0 +1,163 @@
+import { makeTwigInclude } from '../../make-twig-include';
+import { useEffect } from '@storybook/client-api';
+import buttonSwap from './button-swap.twig';
+import { initButtonSwap } from './button-swap.ts';
+// Helper function to initialize toggling button JS
+const buttonSwapStory = (args) => {
+  useEffect(() => {
+    const buttonSwapEls = [
+      ...document.querySelectorAll('.js-c-button-swap'),
+    ].map((buttonSwapEl) =>
+      initButtonSwap(buttonSwapEl, {
+        firstBtnCallback: () => console.log('Subscribed!'),
+        secondBtnCallback: () => console.log('Unsubscribed!'),
+      })
+    );
+    return () =>
+      buttonSwapEls.forEach((buttonSwapEl) => buttonSwapEl.destroy());
+  });
+  return buttonSwap(args);
+};
+
+<Meta
+  title="Components/Button Swap"
+  argTypes={{
+    content_start_icon: {
+      type: { name: 'string' },
+      control: {
+        type: 'select',
+        options: ['bell', 'magnifying-glass'],
+      },
+    },
+    first_button_visually_hidden_text: { type: { name: 'string' } },
+    first_button_label: { type: { name: 'string' } },
+    second_button_visually_hidden_text: { type: { name: 'string' } },
+    second_button_label: { type: { name: 'string' } },
+  }}
+/>
+
+# Button Swap
+
+The Button Swap component was built with the article Push Notifications use case in mind. It has two buttons that are swapped on click representing two visual states accompanied with visually hidden text for a more inclusive experience.
+
+You have the ability to pass `firstBtnCallback` and `secondBtnCallback` callback functions to the `initButtonSwap` initialization function to add custom actions for when each of the buttons is clicked.
+
+<Canvas>
+  <Story
+    name="Default"
+    inline={false}
+    args={{
+      first_button_visually_hidden_text: 'Notifications have been turned off.',
+      first_button_label: 'Give me all the notifications!',
+      second_button_visually_hidden_text: 'Notifications have been turned on.',
+      second_button_label: 'Just kidding, stop notifications, please.',
+    }}
+    parameters={{
+      docs: {
+        source: {
+          code: makeTwigInclude(
+            '@cloudfour/components/button-swap/button-swap.twig',
+            {
+              first_button_visually_hidden_text:
+                'Notifications have been turned off.',
+              first_button_label: 'Give me all the notifications!',
+              second_button_visually_hidden_text:
+                'Notifications have been turned on.',
+              second_button_label: 'Just kidding, stop notifications, please.',
+            }
+          ),
+        },
+      },
+    }}
+  >
+    {(args) => buttonSwapStory(args)}
+  </Story>
+</Canvas>
+
+## Template Properties
+
+- `class` (string): Appends a CSS class(es) to the root element.
+- `content_start_icon` (string, default `'bell'`): The button [icon](/docs/design-icons--page) (the same icon is applied to both buttons).
```

Comment:
Would it make more sense to default to no icon? Or is an icon required? 🤔

Reply:
Hmm...good question. I'll have to think about that. 🤔

Reply:
Thanks for the suggestion, @Paul-Hebert!

After chatting with you on a call and realizing making this change means potentially modifying Buttom template logic, I agree, this doesn't feel like a blocker. Our only use case for this pattern expects a bell icon. I don't see any use cases in the immediate future that would exclude the icon, therefore, I'll leave it as-is for now.

If we _do_ find a new use case where we need this pattern without an icon, we can add in the feature at that point. 😉

Reply:
Sounds good to me :+1: